### PR TITLE
Shaoqz/refactor mutate stru

### PR DESCRIPTION
**Pull request by shaoqx**
Basically finished mutation module refactoring:
- Finished tleap as engine of mutate_stru
- change Mutation name tuple to a class, added many methods
- add several function for List[Mutation] as a representation for mutant
- change run_command to match with functions in old_enzy_htp (mainly the retry)
- (TODO) Partially finished the **topology check** of mutated structure. (Need to finish init_connect of Structure)

Also contain some changes for the MultiEFdesign project
- mainly about assign_mutant function

### CODE REVIEW

**Comment by chrisjurich** [comment]
Code logic looks good. Primary thoughts are on need for:
+ improved docstrings: In general they are not very descriptive and it is hard to tell what they actually do
+ shorter function names: many function names are very long but do not provide much meaning

Please make changes and re-submit.

**Review by shaoqx — COMMENTED** [review]
Added/elaborated more docstring.
changed general.py which contains APIs to api.py
(some general.py is meant to be general as the function in it will distribute to specific files like "residue.py" or "math.py" in the future)

**Comment by chrisjurich** [comment]
Changes look good and are a step in the right direction. Will request more documentation when new release occurs.

**Review by chrisjurich — APPROVED** [review]
Changes are a good step in the right direction. Expect more documentation when final merge into develop/new release occurs.

File: enzy_htp/core/env_manager.py [hunk 1]
```diff
@@ -6,15 +6,18 @@
...
-from subprocess import run
+import time
+from typing import List, Union
+from subprocess import CompletedProcess, SubprocessError, run
+
+from enzy_htp.core.general import get_localtime
 
 from .logger import _LOGGER
-from .exception import MissingEnvironmentElement
+from .exception import EnvMissingExecutable, MissingEnvironmentElement
```
[the remark this answers is not in the document]
Comment (shaoqx) [reply enzy_htp/core/env_manager.py:16]: Good point

File: enzy_htp/core/general.py
[the remark this answers is not in the document]
Comment (shaoqx) [reply enzy_htp/core/general.py:1]: How is it different from general.py tho?

File: enzy_htp/mutation/mutation.py [hunk 2]
```diff
@@ -19,27 +21,161 @@
 import enzy_htp.chemical as chem
 import enzy_htp.structure as es
 
-Mutation = namedtuple("Mutation", "orig target chain_id res_idx")
-Mutation.__doc__ = f"""Named tuple representing a single point mutation in an enzyme.
-   
-
+class Mutation:
```
[the remark this answers is not in the document]
Comment (shaoqx) [reply enzy_htp/mutation/mutation.py:24]: If there are no if_ prefix, how can you determine whether it is switch to some action or an input data just base on the name?

File: enzy_htp/mutation/__init__.py [hunk 3]
```diff
@@ -6,21 +6,16 @@
 Date: 2022-06-15
 """
 
-from .engine import mutate_pdb
-
 from .mutation import (
     Mutation,
     generate_from_mutation_flag,
-    is_valid_mutation,
```
[the remark this answers is not in the document]
Comment (shaoqx) [reply enzy_htp/mutation/__init__.py:14]: It is now a method of Mutation. Maybe change the way of calling it after your next PR?

File: enzy_htp/preparation/protonate.py [hunk 4]
```diff
@@ -270,16 +275,12 @@ def protonate_ligand_with_pybel(stru: Structure,
...
+        fs.clean_temp_file_n_dir([int_ligand_file_path, int_pybel_file_path])
+    fs.clean_temp_file_n_dir([int_ligand_file_dir])
 
-    if _LOGGER.level > 10:  # not DEBUG or below
-        fs.safe_rmdir(int_ligand_file_dir,
-                      empty_only=True)  #prevent remove the entire scratch
     return stru
 
-
+# PYBEL interface
```
[the remark this answers is not in the document]
Comment (shaoqx) [reply enzy_htp/preparation/protonate.py:283]: Yes, Me and Sebastian is working on a new engine rn. This function can always stay as an option.

File: enzy_htp/_interface/pymol_interface.py
[the remark this answers is not in the document]
Comment (shaoqx) [reply enzy_htp/_interface/pymol_interface.py:1]: I'm using it a lot for assign_mutant and structure_selection.

[the remark this answers is not in the document]
Comment (shaoqx) [reply enzy_htp/_interface/pymol_interface.py:1]: What do you mean by owned? We can see how to merge those changes in your PR.

**Pull request merged**